src/server/pmix_server_get.c

```diff
@@ -404,6 +413,29 @@ static pmix_status_t _satisfy_request(pmix_nspace_t *nptr, int rank,
      * having been committed */
     htptr = hts;
     PMIX_CONSTRUCT(&pbkt, pmix_buffer_t);
+
+    /* if they are asking about a rank from an nspace different
+     * from their own, then include a copy of the job-level info */
+    if (NULL != cd &&
+        0 != strncmp(nptr->nspace, cd->peer->info->nptr->nspace, PMIX_MAX_NSLEN)) {
+        cur_rank = PMIX_RANK_WILDCARD;
+        if (PMIX_SUCCESS != (rc = pmix_bfrop.pack(&pbkt, &cur_rank, 1, PMIX_INT))) {
+            PMIX_ERROR_LOG(rc);
+            PMIX_DESTRUCT(&pbkt);
+            cbfunc(rc, NULL, 0, cbdata, NULL, NULL);
+            return rc;
+        }
+        /* the client is expecting this to arrive as a byte object
+         * containing a buffer, so package it accordingly */
+        pbptr = &nptr->server->job_info;
+        if (PMIX_SUCCESS != (rc = pmix_bfrop.pack(&pbkt, &pbptr, 1, PMIX_BUFFER))) {
+            PMIX_ERROR_LOG(rc);
```

Comment:
Both points are valid. However, packing a buffer is really cheap and so this may not be a significant issue? If you feel it is, I guess I could put a flag in there to indicate that we need the job info and attach it to the end of the rank info, if found.

I can/will fix the memory leak.